src/Misc/baseError.ts

```diff
@@ -1,46 +1,66 @@
 import { ErrorCodesType } from "./errorCodes";
 
 /**
- * BabylonJS error internal class wrapper. Extending Error for a class with docs
- * fails typedoc validation, because lib.es5.d.ts does not have any docs for Error.
+ * See https://stackoverflow.com/questions/12915412/how-do-i-extend-a-host-object-e-g-error-in-typescript
+ * and https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work
+ *
+ * Polyfill for Object.setPrototypeOf if necessary.
  * @ignore
  */
-class BabylonJSErrorWrapper extends Error {}
+export const setPrototypeOf: (o: any, proto: object | null) => any =
+ (Object as any).setPrototypeOf || ((o, proto) => { o.__proto__ = proto; return o; });
 
 /**
- * Base Error class for babylonjs errors
+ * Base error. Due to limitations of typedoc-check and missing documentation
+ * in lib.es5.d.ts, cannot extend Error directly for RuntimeError.
+ * @ignore
  */
-export abstract class BaseError extends BabylonJSErrorWrapper {
-    // See https://stackoverflow.com/questions/12915412/how-do-i-extend-a-host-object-e-g-error-in-typescript
-    // and https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work
-
-    // Polyfill for Object.setPrototypeOf if necessary.
-    protected static _setPrototypeOf: (o: any, proto: object | null) => any =
-        (Object as any).setPrototypeOf || ((o, proto) => { o.__proto__ = proto; return o; });
+abstract class BaseError extends Error {}

```

Comment:
If this is staying, might as well put the `setPrototypeOf` back here.

Reply:
I put it out so we can remove BaseError to be exposed alltogether. So that when we fix the Error documentation issue we can remove it cleanly. Lemme know if you feel strongly about this.

Reply:
Reverted it back. When we resolve the Error documentation issue may be do this at that time.